src/main/java/calculator/StringCalculator.java

```diff
@@ -3,51 +3,44 @@
 import org.springframework.util.StringUtils;
 
 import java.util.Arrays;
+import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 /**
  * @author Geonguk Han
  * @since 2020-02-07
  */
 public class StringCalculator {
+
     private static final Pattern CUSTOM_PATTERN = Pattern.compile("//(.)\n(.*)");
+    private static final String DEFAULT_DELIMITER = ",|;";
 
-    public int add(final String stringValue) {
-        if (StringUtils.isEmpty(stringValue)) {
+    public int add(final String input) {
+        if (StringUtils.isEmpty(input)) {
             return 0;
         }
 
-        final Matcher matcher = CUSTOM_PATTERN.matcher(stringValue);
+        final Matcher matcher = CUSTOM_PATTERN.matcher(input);
         if (matcher.find()) {
             final String delimiter = matcher.group(1);
             final String[] split = matcher.group(2).split(delimiter);
-            return calculateProcess(split);
-        }
-
-        final String[] split = stringValue.split(",|;");
-        return calculateProcess(split);
-    }
-
-    private static int calculateProcess(final String[] stringNumbers) {
-        if (isNegative(stringNumbers)) {
-            throw new RuntimeException();
+            Numbers numbers = parseToNumbers(split);
+            return numbers.sum();
         }
 
-        return calculateSum(stringNumbers);
+        final String[] split = input.split(DEFAULT_DELIMITER);
+        Numbers numbers = parseToNumbers(split);
+        return numbers.sum();
     }
 
-    private static boolean isNegative(final String[] stringNumbers) {
-        return Arrays.stream(stringNumbers)
-                .mapToInt(Integer::parseInt)
-                .anyMatch(value -> value < 0);
-    }
+    private Numbers parseToNumbers(final String[] numbers) {
+        List<Number> collect = Arrays.asList(numbers)
+                .stream()
+                .map(s -> new Number(s))
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
                .map(Number::new)
```